make_version.py

```diff
@@ -7,7 +7,15 @@ def main():
     if len(output) == 3:
         version, build, commit = output
     else:
-        raise Exception("Could not git describe, (got %s)" % output)
+        # If the clone is shallow, describe output won't have tag and
+        # number of commits.  This is a particular issue on Travis-CI,
+        # which by default clones with a depth of 50.
+        # This behaviour isn't well documented in git-describe docs,
+        # but see, e.g., https://stackoverflow.com/a/36389573/1008142
+        # and https://github.com/travis-ci/travis-ci/issues/3412
+        version = 'unknown'
+        build = 'unknown'
+        commit = output
```

Comment:
I performed the merge because changes from this PR are needed now, and we can continue discussion and possibly change the version string generation as a separate issue or pull request.